unity-renderer/Assets/UIComponents/Scripts/Components/GridContainer/GridContainerComponentModel.cs

```diff
@@ -1,13 +1,15 @@
 using System;
 using UnityEngine;
+using UnityEngine.Serialization;
 using static UnityEngine.UI.GridLayoutGroup;
 
 [Serializable]
 public class GridContainerComponentModel : BaseComponentModel
 {
     public Constraint constraint = Constraint.FixedColumnCount;
     public int constraintCount = 3;
-    public bool adaptItemSizeToContainer = false;
+    [FormerlySerializedAs("adaptItemSizeToContainer")] public bool adaptHorizontallyItemSizeToContainer = false;
+    public bool adaptVerticallyItemSizeToContainer = false;
```

Comment:
Please take a look to the implementation of the `adaptVerticallyItemSizeToContainer` because it is not working correctly. When you set a grid as `adaptVerticallyItemSizeToContainer = true` and you click on [REFRESH] several times, the height of the grid changes indefinitely and it is not being adapted to the correct height.

![image](https://user-images.githubusercontent.com/64659061/140925824-e3989960-88b8-4e99-9be3-e15bf114adbf.png)

Reply:
Right now this is a TODO since I am only managing when you adapt Horizontally and Vertically, the fixes for this is coming in the next PR